ledger/narwhal/subdag/src/lib.rs

```diff
@@ -78,6 +79,36 @@ fn sanity_check_subdag_with_dfs<N: Network>(subdag: &BTreeMap<u64, IndexSet<Batc
     &commit == subdag
 }
 
+/// Returns the weighted median timestamp of the given timestamps and stakes.
+fn weighted_median(timestamps_and_stake: Vec<(i64, u64)>) -> i64 {
+    let mut timestamps_and_stake = timestamps_and_stake;
+
+    // Sort the timestamps.
+    #[cfg(not(feature = "serial"))]
+    timestamps_and_stake.par_sort_unstable_by_key(|(timestamp, _)| *timestamp);
+    #[cfg(feature = "serial")]
+    timestamps_and_stake.sort_unstable_by_key(|(timestamp, _)| *timestamp);
```

Comment:
You could make [another one of these macros](https://github.com/AleoHQ/snarkVM/blob/testnet3/utilities/src/parallel.rs#L145), more in line with existing code (I only know it from the algorithms crate)